test/test_serialization.h

```diff
@@ -529,7 +529,7 @@ TEST(serialization, sequential_model) {
   network<sequential> net1, net2;
 
   net1 << fully_connected_layer(10, 16) << tanh_layer(16)
-       << average_pooling_layer<relu>(4, 4, 1, 2)
+       << average_pooling_layer(4, 4, 1, 2) << relu_layer(2, 2, 1)
```

Comment:
Maybe it's out of the scope of this PR but as for `network::layer_size` I propose renaming the function name since at least to me it's ambiguous. You can define the size of a network from a very different perspectives, so I think that `network::num_layers` or similar would fit more with the functionality.